internal/engine/buildcontroller.go

```diff
@@ -234,16 +229,11 @@ func buildStateSet(ctx context.Context, manifest model.Manifest, specs []model.T
 		result[id] = buildState
 	}
 
-	// If there are no files changed across the entire state set, then this is a force update.
-	// We want to do an image build of each image.
-	// TODO(maia): I think that instead of storing this on every build state, we can can figure
-	//  out that it's a force update when creating the BuildEntry (in `needsBuild`), store that
-	//  as the BuildReason, and pass the whole BuildEntry to the builder (so the builder can
-	//  know whether to skip in-place builds)
-	// if we're on a crash rebuild, then there won't have been any files changed for that reason
-	if !ms.NeedsRebuildFromCrash && !anyFilesChangedSinceLastBuild {
+	// Check to see if this is a resource in auto mode with an overriding trigger.
+	// See comments on IsOverrideTrigger for more details on the intended flow.
+	if manifest.TriggerMode == model.TriggerModeAuto && ms.TriggerReason != 0 {
```

Comment:
a manual resource may be forced too -- you'd want to check that it was triggered and had no pending file changes.

also fwiw, `if ms.TriggerReason != 0` seems (at least to me) a confusing way to check if the TriggerReason was unset. consider hiding this logic away in a `WasTriggered` (or something) helper func?